lib/sneakers_handlers/retry_handler.rb

```diff
@@ -67,18 +67,20 @@ def create_dlx(channel, queue, options)
       @dlx_queue.bind(dlx_exchange, routing_key: arguments.fetch("x-dead-letter-routing-key"))
     end
 
-    def retry_message(hdr, props, msg)
+    def retry_message(hdr, props, msg, reason)
       headers = props[:headers] || {}
       retry_count = headers["x-retry-count"] || 1
       if retry_count >= @max_retry
         @channel.reject(hdr.delivery_tag)
       else
         Sneakers.logger.info do
-          "Retrying message: queue=#{@queue.name} retry_count=#{retry_count}."
+          "Retrying message: queue=#{@queue.name} retry_count=#{retry_count} reason=#{reason}."
         end
+
+        headers = headers.merge({ "x-retry-count": retry_count + 1, "rejection_reason": reason.to_s })
```

Comment:
is `retry_count` indexed at 0?

Reply:
It's [set](https://github.com/alphasights/sneakers_handlers/pull/27/files#diff-a7624c62df58f71670816eae4bf92daeR72) to `1` in the first retry.